stl/src/vector_algorithms.cpp

```diff
@@ -884,7 +884,8 @@ namespace {
                             _BitScanForward(&_H_pos, _Mask); // lgtm [cpp/conditionallyuninitializedvariable]
 
                             const auto _V_pos = _Traits::_Get_v_pos(_Cur_idx_min, _H_pos); // Extract its vertical index

```

Comment:
Wouldn't it be nicer to have `size_t` type here and avoid `static_cast` below?

Reply:
We could do that if you like, but then I think it would merit a comment that the implicit conversion is absolutely critical, since someone might be tempted to "improve" it back to `auto`. In contrast, the `static_cast` makes it clear that we need the conversion, and the immediate context of the multiplication and additions explains why, without the need for a comment.

Reply:
ok, agree

Reply:
Can we make `size_t{_V_pos}` then instead?

Reply:
Sure! It's sign-preserving, so there's no concern about narrowing conversions, and I think that's sufficiently explicit.

Reply:
Sorry, I forgot about x86, where this can truncate from `unsigned long long` to `size_t`. Have to go back to `static_cast`.